dnf5-plugins/needs_restarting_plugin/needs_restarting.cpp

```diff
@@ -88,6 +114,48 @@ void NeedsRestartingCommand::configure() {
         libdnf5::Option::Priority::RUNTIME, metadata_types);
 }
 
+time_t NeedsRestartingCommand::get_proc_1_boot_time(Context & ctx) {
+    time_t proc_1_boot_time = 0;
+    struct stat proc_1_stat = {};
+    const auto & logger = ctx.get_base().get_logger();
+
+    if (stat("/proc/1", &proc_1_stat) == 0) {
+        proc_1_boot_time = proc_1_stat.st_mtime;
+    } else {
+        logger->debug("Unable to stat /proc/1, using {} as the /proc/1 boot time.", proc_1_boot_time);
+    }
+
+    return proc_1_boot_time;
+}
+
+time_t NeedsRestartingCommand::get_kernel_boot_time(Context & ctx) {
+    time_t btime = 0;
+    const auto & logger = ctx.get_base().get_logger();
+    std::ifstream proc_stat{"/proc/stat"};
+    std::string line;
+
+    if (!proc_stat.is_open()) {
+        logger->debug("Unable to read /proc/stat, using {} as the system boot time.", btime);
+        return btime;
+    }
+
+    while (std::getline(proc_stat, line)) {
+        if (line.compare(0, 6, "btime ") == 0) {
+            std::istringstream btime_stream(line);
+            std::string tmp;
+            btime_stream >> tmp >> btime;
+            return btime;
+        }
+    }
+
+    if (btime == 0) {
+        logger->debug("Error reading /proc/stat, using {} as the system boot time.", btime);
+    }
+
+    logger->debug("Using {} as the system boot time.", btime);
```

Comment:
Let's remove this line

Reply:
By this I assume you mean line 155, in which case done.